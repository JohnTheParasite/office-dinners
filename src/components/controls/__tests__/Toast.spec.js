import { createLocalVue, mount, shallowMount } from "@vue/test-utils"
import toast from "@/components/controls/Toast"
import i18n from "@/i18n"

describe("Toast with timeouts", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)

  it("toast render", async () => {
    jest.useFakeTimers()
    jest.spyOn(global, "setTimeout")

    let toastElem = await shallowMount(toast, { localVue, i18n })

    expect(setTimeout).toHaveBeenCalledTimes(1)
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 5000)

    jest.runOnlyPendingTimers()

    expect(toastElem).toMatchSnapshot()
  })

  it("leave(event) should work", () => {
    jest.useFakeTimers()
    jest.spyOn(global, "setTimeout")

    const component = {
      template: "<div><toast></toast></div>",
      components: { toast }
    }

    let mockfn = jest.fn()

    document.body.removeChild = mockfn

    let divElem = mount(component, { localVue, i18n })

    let toastElem = divElem.findComponent(toast)

    toastElem.vm.leave(toastElem.vm.$el)

    expect(setTimeout).toHaveBeenCalledTimes(1)
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 250)

    jest.runOnlyPendingTimers()

    expect(mockfn).toBeCalled()
  })
})

describe("Tosts", () => {
  const localVue = new createLocalVue()
  localVue.use(i18n)

  it("should render toast with computed", async () => {
    let toastElem = await shallowMount(toast, { localVue, i18n })

    expect(toastElem).toMatchSnapshot()
  })

  it("should render toast with name", async () => {
    let toastElem = await shallowMount(toast, { localVue, i18n, propsData: { header: "test" } })

    expect(toastElem).toMatchSnapshot()
  })
})
