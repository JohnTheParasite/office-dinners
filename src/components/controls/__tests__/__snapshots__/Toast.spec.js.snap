// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast toast render 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;

exports[`Toast toast render with computed 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;

exports[`Toast toast render with name 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;

exports[`Toast with timeouts toast render 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;

exports[`Tosts should render toast with computed 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;

exports[`Tosts should render toast with name 1`] = `
<transition-stub name="toast">
  <!---->
</transition-stub>
`;
